script/create_changelog

```diff
@@ -0,0 +1,402 @@
+   - repo-name/changelogs/released/tag-name (new directory)
+
+- Pushing branch to repo
+   To push the changes to github, script will add new remote, named upstream,
+   using your USERNAME and TOKEN
+
+- Creating a pull request
+   Script uses github API to create a pull request, using your USERNAME and TOKEN.
+   Refer https://developer.github.com/v3/pulls/#create-a-pull-request for API
+EOF
+}
+
+#is_stable_tag check if given tag=$1 is stable tag or not
+#if tag is v1.1.0 then it will return 0
+#if tag is v1.1.0-RC1 then it will return 1
+is_stable_tag() {
+	[[ "$(echo $1 | cut -d '-' -f1)" == "$1" ]] && echo 0 || echo 1
+}
+
+#is_rc_tag check if given tag=$1 is RC tag or not
+# if tag is v1.1.0-RC1 then it will return 0
+# if tag is v1.1.0-custom-RC1 then it will return 1
+# if tag is v1.1.0-RC1A then it will return 2
+is_rc_tag() {
+        local suffix=$(echo $1 | cut -d '-' -f2-)
+        local suffix_str=${suffix::2}
+        local suffix_num=${suffix:2}
+
+        [[ "$suffix_str" != "RC" ]] && echo 1 && return
+        [[ "$suffix_num" == "" ]] && echo 2 && return
+        [[ $suffix_num =~ ^[0-9]+$ ]] && (echo 0 && return) || echo 2
+}
+
+#is_tag_belongs_to_release check if tag=$2 is part of release=$1
+# if tag is v1.1.0-RC1 and release is v1.1.0 then it will return 0
+# if tag is v1.0.0-RC1 and release is v1.1.0 then it will return 1
+# if tag is v1.1.0-custom-RC1 and release is v1.1.0 then it will return 1
+is_tag_belongs_to_release() {
+	release=$1
+	tag=$2
+
+	[[ -z $release ]] && echo 1 && return
+	[[ -z $tag ]] && echo 1 && return
+
+	[[ "$(echo $tag | cut -d '-' -f1)" == $release ]] && echo 0 || echo 1
+}
+
+#add_changelog add the changelog from files in folder=$1 to file=$2
+add_changelog() {
+	cl_dir=$1
+	cl_file=$2
+
+	[[ -d $cl_dir ]] || return
+	[[ -f $cl_file ]] || return 
+
+	echo "Adding changelog from $cl_dir"
+
+	for i in `find $cl_dir -type f`; do
+		pr_number=$(echo $i | cut -d  '/' -f4 |awk -F '-' '{print $1}')
+		usr_name=$(echo $i | cut -d  '/' -f4 |awk -F '-' '{print $2}')
+		msg=$(cat $i)
+
+		url_usr=https://github.com/$usr_name
+		url_pr=https://github.com/$GIT_REPO/pull/$pr_number
+		echo "* $msg ([#$pr_number]($url_pr),[@$usr_name]($url_usr))" >> $cl_file
+		echo -n "- $url_pr\r\n" >>  $pr_msg_file
+	done
+}
+
+#generate_changelog create the tag directory under $RELEASED_CHANGELOG_DIR
+#and move all files from $UNRELEASED_CHANGELOG_DIR to new tag directory
+#It also updates the $ROOT_CHANGELOG with tag changelog
+generate_changelog() {
+	echo "Generating Changelog for $TAG"
+	local dest=$RELEASED_CHANGELOG_DIR/$TAG
+
+	[[ -d $CHANGELOG_DIR ]] || return 3
+	[[ -d $UNRELEASED_CHANGELOG_DIR ]] || mkdir $UNRELEASED_CHANGELOG_DIR
+	[[ -d $RELEASED_CHANGELOG_DIR ]] || mkdir $RELEASED_CHANGELOG_DIR
+	[[ -d $dest ]] || mkdir $dest
+
+	ls $UNRELEASED_CHANGELOG_DIR/* > /dev/null 2>&1 && cp $UNRELEASED_CHANGELOG_DIR/* $dest
+
+	cl=$(mktemp)
+	echo "$TAG / $(date +'%Y-%m-%d')" >> $cl
```

Comment:
we are taking the date when we are creating the changelog, not when the tag was created.@kmova should it be date when tag was created or date when we are running the changelog?

Reply:
updating the changelog directly might result in merge conflicts in parallel PRs.